components/tools/OmeroPy/src/omero/gateway/__init__.py

```diff
@@ -2605,17 +2599,14 @@ def getGroupsMemberOf(self):
         :rtype:         :class:`ExperimenterGroupWrapper` generator
         """
 
-        q = self.getQueryService()
-        p = omero.sys.Parameters()
-        p.map = {}
-        p.map["ids"] = rlist([rlong(a)
-                              for a in self.getEventContext().memberOfGroups])
-        sql = "select e from ExperimenterGroup as e where e.id in (:ids)"
-        for e in q.findAllByQuery(sql, p, self.SERVICE_OPTS):
-            if e.name.val == "user":
-                pass
-            else:
-                yield ExperimenterGroupWrapper(self, e)
+        groups = list()
+        system_groups = [
+            self.getAdminService().getSecurityRoles().userGroupId]
+        for g in self.getObjects("ExperimenterGroup",
+                                 self.getEventContext().memberOfGroups):
+            if g.getId() not in system_groups:
+                groups.append(g)
+        return groups
```

Comment:
Do we wnt to use ``` yield ``` to return a generator here instead of a list, to be consistent with getObjects() and getGroupsLeaderOf()?